pkg/apis/resource/helper.go

```diff
@@ -1,11 +1,188 @@
 package resource
 
 import (
+	"bytes"
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+
 	"github.com/seal-io/walrus/pkg/dao/model"
+	"github.com/seal-io/walrus/pkg/dao/model/resource"
 	"github.com/seal-io/walrus/pkg/dao/types/object"
 	"github.com/seal-io/walrus/pkg/dao/types/property"
+	pkgresource "github.com/seal-io/walrus/pkg/resource"
+	pkgvariable "github.com/seal-io/walrus/pkg/variable"
+	"github.com/seal-io/walrus/utils/json"
 )
 
+var (
+	// _variableReg the regexp to match the variable.
+	variableReg = regexp.MustCompile(`\${var\.([a-zA-Z0-9_-]+)}`)
+	// _resourceReg the regexp to match the resource output.
```

Comment:
There are two regexp `\${res\.([^.}]+)\.([^.}]+)` and `\${var\.([a-zA-Z0-9_-]+)}` in this package and `pkg/deployer/terraform` can we unify them? Please update the comment of the variable.